pkg/networkservicemesh/common/setid/client.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2020 Cisco Systems, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setid
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservicemesh/core/next"
+)
+
+type idClient struct {
+	name string
+}
+
+// NewClient - creates a new setId client.
+//             name - name of the client
+//             Iff the current pathSegment name != name && pathsegment.id != connection.Id, set a new uuid for
+//             connection id
+func NewClient(name string) networkservice.NetworkServiceClient {
+	return &idClient{name: name}
+}
+
+func (i *idClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*connection.Connection, error) {
```

Comment:
We take an `opts ...grpc.CallOption` but don't do anything with this. Is this intentional?

Reply:
Nope... should have passed it through to next.Client(ctx)... fixed.